python/ray/experimental/workflow/api.pyi

```diff
@@ -0,0 +1,78 @@
+# yapf: disable
+from typing import Callable, Generic, Optional, TypeVar, Union, overload
+
+from ray._raylet import ObjectRef
+
+T0 = TypeVar("T0")
+T1 = TypeVar("T1")
+T2 = TypeVar("T2")
+T3 = TypeVar("T3")
+T4 = TypeVar("T4")
+T5 = TypeVar("T5")
+T6 = TypeVar("T6")
+T7 = TypeVar("T7")
+T8 = TypeVar("T8")
+T9 = TypeVar("T9")
+R = TypeVar("R")
```

Comment:
I'm not sure whether it's the best way to do it in python.  I'm also wondering whether we can do it better.
```
Ts = [TypeVar(f"T{i}" for i in range(10))]
ArgTs = [Union[T, ObjectRef[T], Workflow[T]] for T in Ts]
```

So later, we can use 
```
class WorkflowStepFunction(Generic[Ts, R]):
    def __init__(self, function: Callable[Ts, R]): ...
    
    @overload
    def step(self, arg0: ArgTs[0]): ...
```

Reply:
Unfortunately type checking softwares cannot understand this. I learned this syntax from `worker.pyi` in ray.